plugins/inputs/mongodb/mongodb.go

```diff
@@ -78,118 +81,104 @@ func (*MongoDB) Description() string {
 	return "Read metrics from one or many MongoDB servers"
 }
 
-var localhost = &url.URL{Host: "mongodb://127.0.0.1:27017"}
+func (m *MongoDB) Init() error {
+	var tlsConfig *tls.Config
+	if m.Ssl.Enabled {
+		// Deprecated TLS config
+		tlsConfig = &tls.Config{}
+		if len(m.Ssl.CaCerts) > 0 {
+			roots := x509.NewCertPool()
+			for _, caCert := range m.Ssl.CaCerts {
+				if ok := roots.AppendCertsFromPEM([]byte(caCert)); !ok {
+					return fmt.Errorf("failed to parse root certificate")
+				}
+			}
+			tlsConfig.RootCAs = roots
+		} else {
+			tlsConfig.InsecureSkipVerify = true
```

Comment:
I see this isn't new code, but I'm not a fan of subtly letting the user enable SSL but then quietly skipping certs and connecting, as it's vulnerable to MITM (man in the middle) attacks. I think we've explicitly made the user set `insecure_skip_verify = true` in other places.